release-notes.txt

```diff
@@ -1,3 +1,548 @@
+RIOT-2024.04 - Release Notes
+============================
+RIOT is a multi-threading operating system which enables soft real-time
+capabilities and comes with support for a range of devices that are typically
+found in the Internet of Things: 8-bit and 16-bit microcontrollers as well as
+light-weight 32-bit processors.
+
+RIOT is based on the following design principles: energy-efficiency, soft
+real-time capabilities, small memory footprint, modularity, and uniform API
+access, independent of the underlying hardware (with partial POSIX compliance).
+
+RIOT is developed by an international open-source community which is
+independent of specific vendors (e.g. similarly to the Linux community) and is
+licensed with a non-viral copyleft license (LGPLv2.1), which allows indirect
+business models around the free open-source software platform provided by RIOT.
+
+
+About this release
+==================
+
+The 2024.04 release includes:
+
+## List of notable changes here ##
+
+210 pull requests, composed of 446 commits, have been merged since the
+last release, and 4 issues have been solved. 42 people contributed with
+code in 82 days. 2409 files have been touched with 26218 (+) insertions and
+29933 deletions (-).
+
+
+Notations used below
+====================
+
+    + means new feature/item
+    * means modified feature/item
+    - means removed feature/item
+
+
+New features and changes
+========================
+
+Core
+----
+
++ core/lib/cib: add new functionality (#20464)
++ macros/utils: add LIMIT() and ABS() macros (#20352)
+* mbox: provide function to unset initialized mbox (#15485)
+
+System Libraries
+----------------
+
++ add error check of aes_encrypt() (#15576)
++ event_periodic_callback: add event_periodic_callback_create() (#20459)
++ examples/rust-gcoap: Add SAUL (#17554)
++ shell/cmds: add udptty command (#19742)
++ sys/libc: add endian.h (#20310)
++ sys/shell/lwip: add IPv4 configuration to lwip ifconfig command (#19849)
++ tests: Add NULL pointer argument test for fmt_lpad/to_lower/str (#20344)
+* makefiles/cargo: Remove CARGO_CHANNEL and other simplifications (#20319)
+* Monitoring malloc usage (#20363)
```

Comment:
This is certainly not a "big fish". But it is super handy and IMO something people should be aware of, in case they need to trace down issues with dynamic memory management.

Reply:
I added it to the highlights.